src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageModelActivatorProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    /// <summary>
+    /// <see cref="IPageActivatorProvider"/> that uses type activation to create Pages.
+    /// </summary>
+    public class DefaultPageModelActivatorProvider : IPageModelActivatorProvider
```

Comment:
We'll also need the equivalent plumbing for the DI versions of these. Please open an issue for that or do it right away 👍